packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -206,19 +207,15 @@ library LibProving {
                 revert L1_INVALID_PROOF();
             }
 
-            (bool verified, bytes memory ret) = resolver.resolve(
-                LibUtils.getVerifierName(evidence.verifierId), false
-            ).staticcall(evidence.proof);
-
-            if (
-                !verified
-                //
-                || ret.length != 32
-                //
-                || bytes32(ret) != keccak256("taiko")
-            ) {
-                revert L1_INVALID_PROOF();
-            }
+            // Later on it can be like verifyProofs(bytes) where the bytes would
+            // include all of the proofs (ZK, SGX, Oracle, etc. ).
+            IProofVerifier(resolver.resolve("proof_verifier", false)).verifyProofs(
+                abi.encode(
+                    evidence.verifierId, 
```

Comment:
we should use delete `evidence.verifierId` and `evidence.proof` from evidence and add a new field called `bytes proofs`. By doing this, the protocol will be agnostic to verifierId which is ZK related. Then you just send `proofs` to the prover contract. 

In side the prover contact, you decode `proofs` into a struct and then further use the `verifiedId` to dispatch zkProofs to zkVerifier.

Reply:
In core protocol, no ZK proof info should be exposed.

Reply:
Anything else you want to know? I thought it's pretty clear: making sure the core protocol does not know anything about the content of a block proof and how they shall be verified. Move these logics to the Verifier.